From 4b2863c4c100ab47af11616d050582653fb45d13 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 14 Mar 2020 23:46:20 +0100
Subject: [PATCH 1478/1936] PADnoteParameter fix for QtCreator

Make PADnoteParameters usable with QtCreator again. Necessary due to
QTCREATORBUG-23722 (no capturing of VLAs in lambdas).
---
 src/Params/PADnoteParameters.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index dfdee77c..ceb38925 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -898,12 +898,15 @@ int PADnoteParameters::sampleGenerator(PADnoteParameters::callback cb,
     float adj[samplemax];
     for(int nsample = 0; nsample < samplemax; ++nsample)
         adj[nsample] = (Pquality.oct + 1.0f) * (float)nsample / samplemax;
+    // QtCreator can't capture VLAs (QTCREATORBUG-23722), so this is
+    // a workaround to allow using the IDE
+    float * const adj_ptr = adj;
 
     const PADnoteParameters* this_c = this;
 
     auto thread_cb = [basefreq, bwadjust, &cb, do_abort,
                       samplesize, samplemax, spectrumsize,
-                      &adj, &profile, this_c](
+                      adj_ptr, &profile, this_c](
                       unsigned nthreads, unsigned threadno)
     {
         //prepare a BIG IFFT
@@ -917,7 +920,7 @@ int PADnoteParameters::sampleGenerator(PADnoteParameters::callback cb,
             if(do_abort())
                 break;
             const float basefreqadjust =
-                powf(2.0f, adj[nsample] - adj[samplemax - 1] * 0.5f);
+                powf(2.0f, adj_ptr[nsample] - adj_ptr[samplemax - 1] * 0.5f);
 
             if(this_c->Pmode == 0)
                 this_c->generatespectrum_bandwidthMode(spectrum,
-- 
2.47.0

